core/testcontainers/core/container.py

```diff
@@ -96,16 +144,17 @@ def start(self) -> Self:
             ports=self.ports,
             name=self._name,
             volumes=self.volumes,
+            labels={"hash": hash_},
             **self._kwargs,
         )
         logger.info("Container started: %s", self._container.short_id)
-        if self._network:
-            self._network.connect(self._container.id, self._network_aliases)
-        return self
 
     def stop(self, force=True, delete_volume=True) -> None:
         if self._container:
-            self._container.remove(force=force, v=delete_volume)
+            if self._reuse and c.tc_properties_testcontainers_reuse_enable:
+                self._container.stop()
+            else:
+                self._container.remove(force=force, v=delete_volume)
```

Comment:
hm, isnt the point to not even stop it so it is warm for next run? i guess if people are using the explicit api then whatever. I do see a bit of a mirror with start so i guess it will just have to be consistent and maybe clear in docs.

Reply:
In other languages, having a reusable container *does not* change the contract of the `stop()` method. This is obviously something that needs to be considered to make this a full fledged use case, but as of now, I would suggest we start with an experimental reusable implementation, that mirrors the Java implementation.

Reply:
Okay, I see. I updated the code and documentation (how to use reusable containers) to not change the contract of the `stop()` method.